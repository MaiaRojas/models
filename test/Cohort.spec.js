const mongoose = require('mongoose');
const { Campus, Cohort } = require('../')(mongoose);


describe('Cohort', () => {
  let campus;

  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await Campus.createIndexes();
    await Cohort.createIndexes();
    campus = new Campus({
      slug: 'lim',
      name: 'Lima',
      locale: 'es-PE',
      timezone: 'America/Lima',
      active: true,
    });
    await campus.save();
  });

  it('should fail when campus does not exist', () => {
    const cohort = new Cohort({
      slug: 'lim-2017-09-bc-core-am',
      campus: (new Campus({}))._id,
      program: 'bc',
      track: 'core',
      name: 'am',
      generation: 9,
      start: new Date(),
      end: new Date(),
    });

    return cohort.save()
      .catch(err => expect(err.message).toBe('Campus does not exist'));
  });

  it('should create common core cohort for a given generation', () => {
    const cohort = new Cohort({
      slug: 'lim-2017-09-bc-core-am',
      campus: campus._id,
      program: 'bc',
      track: 'core',
      name: 'am',
      generation: 1,
      start: new Date(),
      end: new Date(),
    });

    return cohort.save()
      .then((result) => {
        const {
          _id,
          campus: cohortCampus,
          start,
          end,
          ...obj
        } = result.toJSON();
        expect(obj).toMatchSnapshot();
        return Cohort.findById(_id);
      })
      .then((doc) => {
        expect(`${doc._id}`).toBe(`${cohort._id}`);
      });
  });
});
