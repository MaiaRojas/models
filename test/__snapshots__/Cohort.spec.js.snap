// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should create L4B cohort 1`] = `
Object {
  "__v": 0,
  "name": "alicorp",
  "program": "l4b",
  "projects": Array [],
  "publicAdmission": false,
  "rubric": "2",
  "topics": Array [],
  "track": "business",
}
`;

exports[`Cohort should create admission cohort (new generation) for existing campus 1`] = `
Object {
  "__v": 0,
  "generation": 2,
  "name": "am",
  "program": "pre",
  "projects": Array [],
  "publicAdmission": false,
  "rubric": "2",
  "topics": Array [],
  "track": "core",
}
`;

exports[`Cohort should create admission cohort (new generation) for new campus 1`] = `
Object {
  "__v": 0,
  "generation": 1,
  "name": "am",
  "program": "pre",
  "projects": Array [],
  "publicAdmission": false,
  "rubric": "2",
  "slug": "lim-2017-09-pre-core-am",
  "topics": Array [],
  "track": "core",
}
`;

exports[`Cohort should create common core cohort for a given generation 1`] = `
Object {
  "__v": 0,
  "generation": 1,
  "name": "am",
  "program": "bc",
  "projects": Array [],
  "publicAdmission": false,
  "rubric": "2",
  "slug": "lim-2017-09-bc-core-am",
  "topics": Array [],
  "track": "core",
}
`;
